source/common/http/http1/codec_impl.cc

```diff
@@ -371,6 +378,7 @@ void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end
   const HeaderEntry* path = headers.Path();
   const HeaderEntry* host = headers.Host();
   bool is_connect = HeaderUtility::isConnect(headers);
+  setIsResponseToConnectRequest(is_connect);
```

Comment:
In this case, isn't it the actual CONNECT request, not the response to CONNECT request?

Reply:
yeah, I was doing an experiment here because I thought that `content_length` and `chunk_encoding` headers should be ignored in this test:

https://github.com/envoyproxy/envoy/blob/da60c3726522138182226e9f0deaa8a1e17da4e8/source/common/http/http1/codec_impl.cc#L268

when the method is CONNECT as mentioned in the RFC, but I was not sure how to convey the method information. If you have any suggestions on how to handle this I'm taker, otherwise, I'll take another look.